articles/142_idl.md

```diff
@@ -191,7 +191,7 @@ The next table defines how IDL types are mapped to the following programming lan
 | float       | float            | float                          | float                |
 | double      | double           | double                         | float                |
 | long double | long double      | long double[<sup>2</sup>](#ld) | float                |
-| char        | unsigned char    | unsigned char                  | str with length 1    |
```

Comment:
The mapping documented here is correct. A `char` in `IDL` is mapped to a `str` with length 1 in Python.

The mapping from the `char` type in `.msg` files is described in [143_legacy_interface_definition](https://github.com/ros2/design/blob/ae61bd9738c0896e97ff4768ffd7b898435f2ff0/articles/143_legacy_interface_definition.md) instead.

Reply:
thanks for clarifying, to summarize:
- the `char` type mapping for IDL is the same as the crystal one for msg
- `char` in msg for crystal will be `int` and not `str`
- the design document that needs to be updated is the one defining the python mapping for msg

Is that correct?

See https://github.com/ros2/design/pull/222 for a second attempt

Reply:
In Crystal and before the `char` type in `.msg` was mapped to a single character (`str` with length 1 in Python). That mapping was different that how it is specified in ROS 1 and therefore as of Dashing the mapping as it was in ROS 1 has been restored which maps a `char` in `.msg` to an `uint8`. See #213.

Reply:
Thanks for the answer :+1: 

My question was mostly find a place to share the information with users (familiar or not with ROS1 mapping) that may face the same issue.
It seems that these documents are not the right place. Where do you think it would be appropriate? maybe on a Dashing migration page? or an answered question of ROS Answers?

Reply:
The dashing release page would be the right place. It already contains a migration section where this information would fit in: https://index.ros.org/doc/ros2/Releases/Release-Dashing-Diademata/#changes-since-the-crystal-release

Reply:
:+1: thanks, submitting a PR there then

Reply:
Already done: ros2/ros2_documentation#149

Reply:
:+1: 
closing https://github.com/ros2/ros2_documentation/pull/148 then

Reply:
Sorry for the duplicate effort. Thanks for following up on this.